clusterBuildStrategy/source-to-image/README.md

```diff
@@ -0,0 +1,118 @@
+# Source-To-Image (S2I) ClusterBuildStrategy
+The `source-to-image` ClusterBuildStrategy is composed of [source-to-image](https://github.com/openshift/source-to-image/) and [buildah](https://github.com/containers/buildah) in order to generate a `Dockerfile` and prepare the application to be built later with a builder-image.
+
+`s2i` requires a specially crafted image, which can be informed as `builder-image` parameter on the `Build` resource.
+
+## Getting Started with Buildah ClusterBuildStrategy
+
+### Pre-requisites:
+- The `Builds for Red Hat OpenShift` Operator is available in the OpenShift cluster.
+- The `source-to-image-redhat` clusterBuildStrategy is already installed. This could be checked by following command:
+```
+$ oc get cbs | grep source-to-image
+```
+If the buildStrategy is not available, [Install ClusterBuildStrategies](../../README.md) first.
+
+### Steps:
+1. Create a builds object. 
+- `source` refers to the location where the source code is.
+- `strategy` references the BuildStrategy to use to build the container.
+- `paramValues` is a list of key/value that could be used to set strategy parameters.
+    - Here, we pass the `builder-image` parameter to specify the builder image to be used to build the output image.
+- `output` refers to the location where the built image would be pushed.
+    - Here, it is expected to be pushed to a custom quay repo.
+    - `pushSecret` is the secret name that stores the credentials for pushing container images. Follow [Creating registry secret](section), if the secret doesn't exist already.
+```
+$ cat <<EOF | oc create -f -     
+apiVersion: shipwright.io/v1beta1
+kind: Build
+metadata:
+  name: s2i-nodejs-build
+spec:
+  source:
+    git:
+      url: https://github.com/shipwright-io/sample-nodejs
+    contextDir: source-build/
+  strategy:
+    name: source-to-image-redhat
+    kind: ClusterBuildStrategy
+  paramValues:
+  - name: builder-image
+    value: "quay.io/centos7/nodejs-12-centos7"
+  output:
+    image: quay.io/repo/s2i-nodejs-example
+    pushSecret: registry-credential
+EOF
+```
+
+2. Ensure that the `pipeline` ServiceAccount is available in the current namespace, and has `privileged` SCC assigned. If not, create one & add the SCC.
+- If missed, the buildRun will report failure with `PodAdmissionFailed` Reason.
+```
+$  oc adm policy add-scc-to-user privileged -z pipeline
```

Comment:
I would remove this part
Did you need this on your test? i dont think so if using the builds operator

Reply:
Yes, I tried this using builds operator & this was needed, because of the `DAC_OVERRIDE` capability required in one of the steps, which isn't available with pipelines-scc by default.

Reply:
Is there an alternate way we can follow to use the `pipelines-scc` but have the `DAC_OVERRIDE` cap?
I am not sure, if we can suggest the tekton team to modify `pipelines-scc` caps.

Reply:
is DAC_OVERRIDE required for buildah ? AFAIK we run now buildah without specific previliges ?
can we try to confirm why we need this ?